django_fido/views.py

```diff
@@ -219,22 +219,27 @@ class Fido2AuthenticationViewMixin(Fido2ViewMixin):
     Ensure user to be authenticated exists.
     """
 
-    def get_user(self: View):
+    def get_user(self: View) -> Optional[AbstractBaseUser]:
         """
         Return user which is to be authenticated.
 
         Return None, if no user could be found.
         """
         user_pk = self.request.session.get(AUTHENTICATION_USER_SESSION_KEY)
-        if user_pk is None:
-            return None
-        return get_user_model().objects.get(pk=user_pk)
+        username = self.request.GET.get('username')
+
+        if SETTINGS.two_step_auth and user_pk is not None:
+            return get_user_model().objects.get(pk=user_pk)
+        if not SETTINGS.two_step_auth and username is not None:
+            return get_user_model().objects.get_by_natural_key(username)
+        return None
 
     def dispatch(self, request, *args, **kwargs):
         """Redirect to login, if user couldn't be found."""
```

Comment:
I would update the docstring to explain the difference in logic for single/two step auth.

Reply:
Fixed.